src/EditorFeatures/CSharpTest/SymbolKey/SymbolKeyCompilationsTests.cs

```diff
@@ -133,8 +133,8 @@ partial void M() { }
         var implementation = definition.PartialImplementationPart;
 
         // Assert that both the definition and implementation resolve back to themselves
-        Assert.Equal(definition, ResolveSymbol(definition, comp, SymbolKeyComparison.None));
-        Assert.Equal(implementation, ResolveSymbol(implementation, comp, SymbolKeyComparison.None));
+        Assert.Same(definition, ResolveSymbol(definition, comp, SymbolKeyComparison.None));
```

Comment:
why werre these changed?  i'm not sure that .Same is actually the guarantee we intend to make with SymbolKey

Reply:
Good point, I'm not sure either what's the guarantee of SymbolKey, I can revert.

Reply:
I don't believe it is 'Same'.  Especially as we're just doing normal symbol lookups on the Compilation instance, and it doesn't give this guarantee.  So let's move back to .Equal as that definitely is something we are guaranteeing.  Thanks!